From 7c9b3bcec82dc223de7096b2206965f4205f1e0d Mon Sep 17 00:00:00 2001
From: Mateusz Jonak <mateusz@web.blue>
Date: Fri, 12 Dec 2025 15:53:28 +0100
Subject: [PATCH 12/51] fix: Reset gas trace counts using dedicated function

This commit updates the gas tracing functionality in the QuickJS runtime by replacing the direct memory reset with a call to the new function js_gas_trace_reset_counts. This change improves code clarity and ensures that gas trace counts are reset correctly, enhancing the overall reliability of the gas tracing system.
---
 quickjs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/quickjs.c b/quickjs.c
index 6b052ab..ac14932 100644
--- a/quickjs.c
+++ b/quickjs.c
@@ -3026,7 +3026,7 @@ int JS_EnableGasTrace(JSContext *ctx, int enabled)
     if (!trace)
         return -1;
 
-    memset(trace, 0, sizeof(*trace));
+    js_gas_trace_reset_counts(trace);
     trace->enabled = enabled ? TRUE : FALSE;
     return 0;
 }
